### PR TITLE
Delete any remaining dangling resorces.

### PR DESCRIPTION
**Pull request by fedepaol**
This includes:
- sriov networks / policies
- ptp labels / configurations
- sctp / sriov namespaces

### CODE REVIEW

File: functests/ptp/ptp.go [hunk 1]
```diff
@@ -168,6 +143,35 @@ var _ = Describe("ptp", func() {
...
+		_, err = client.Client.Nodes().Update(&node)
+		Expect(err).ToNot(HaveOccurred())
+	}
+
+	nodeList, err = client.Client.Nodes().List(metav1.ListOptions{LabelSelector: fmt.Sprintf("%s=", ptpSlaveNodeLabel)})
+	Expect(err).ToNot(HaveOccurred())
+	for _, node := range nodeList.Items {
+		delete(node.Labels, ptpSlaveNodeLabel)
+		_, err = client.Client.Nodes().Update(&node)
+		Expect(err).ToNot(HaveOccurred())
```
[the remark this answers is not in the document]
Comment (fedepaol) [reply functests/ptp/ptp.go:171]: This code was just moved inside a function, we can make it cleaner in a followup PR.

File: functests/ptp/ptp.go [hunk 2]
```diff
@@ -168,6 +143,35 @@ var _ = Describe("ptp", func() {
 	})
 })
 
+// Clean removes the current ptp configuration
+func Clean() {
+	ptpconfigList, err := client.Client.PtpConfigs(ptpLinuxDaemonNamespace).List(metav1.ListOptions{})
+	Expect(err).ToNot(HaveOccurred())
```
[the remark this answers is not in the document]
Comment (fedepaol) [reply functests/ptp/ptp.go:149]: This logic is shared between the test implementation (where it makes sense to bail out at the first failure) and the cleaning logic. The general approach is to scream whenever an error happens, I would leave this this way.

File: functests/test_suite_test.go [hunk 3]
```diff
@@ -98,22 +103,31 @@ var _ = BeforeSuite(func() {
...
 var _ = AfterSuite(func() {
-	err := testclient.Client.Namespaces().Delete(testutils.NamespaceTesting, &metav1.DeleteOptions{})
+	// Needed because sctp default namespace tests create resources
+	// that needs to be selectively deleted
+	err := namespaces.Clean("default", "testsctp-", testclient.Client)
 	Expect(err).ToNot(HaveOccurred())
-	err = namespaces.WaitForDeletion(testclient.Client, testutils.NamespaceTesting, 5*time.Minute)
+	err = clean.SriovResources()
 	Expect(err).ToNot(HaveOccurred())
+	ptp.Clean()
```
[the remark this answers is not in the document]
Comment (fedepaol) [reply functests/test_suite_test.go:116]: Yes, I plan to make everything a bit more omogeneous in 4.6, where each test suite / functionality exports a Clean() method we can rely on.

File: functests/test_suite_test.go [hunk 4]
```diff
@@ -98,22 +103,31 @@ var _ = BeforeSuite(func() {
...
-
-	err = testclient.Client.Namespaces().Delete(namespaces.DpdkTest, &metav1.DeleteOptions{})
-	Expect(err).ToNot(HaveOccurred())
-	err = namespaces.WaitForDeletion(testclient.Client, namespaces.DpdkTest, 5*time.Minute)
-	Expect(err).ToNot(HaveOccurred())
+	nn := []string{testutils.NamespaceTesting,
+		perfUtils.NamespaceTesting,
+		namespaces.DpdkTest,
+		sctp.TestNamespace,
+		sriovNamespaces.Test,
```
[the remark this answers is not in the document]
Comment (fedepaol) [reply functests/test_suite_test.go:122]: Same, already on my list

File: functests/ptp/ptp.go (hunk 2 above)
[the remark this answers is not in the document]
Comment (fromanirh) [reply functests/ptp/ptp.go:149]: good enough for me.

File: functests/ptp/ptp.go (hunk 1 above)
[the remark this answers is not in the document]
Comment (fromanirh) [reply functests/ptp/ptp.go:171]: good enough for me.

**Pull request merged**